### PR TITLE
#164139700 Add ability to like and unlike comments

### PR DESCRIPTION
**Pull request by olajide1234**
#### What does this PR do?
Add ability to like and unlike comments
#### Description of Task to be completed?
- Add like/unlike comment endpoint
- Add`put` method to route
- Implement logic to enable user like the comment if they hit the route for the first time
- If they hit the route again, the comment should be unliked
#### How should this be manually tested?
- run  `npm run db-migrate` to get all database migration updates.
- make a `PUT` request to the `/comment/:id/like` route with the ID of comment to like
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139700

#### Questions:
N/A

### CODE REVIEW

File: helpers/isCommentLikedByUser.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { CommentLikes } from '../models';
+
+/**
+ * @description This function checks if a function has been liked by a user
```
Comment (Dubby20) [inline helpers/isCommentLikedByUser.js]: Correct the mistake in your description js.docs

File: routes/v1.js [hunk 2]
```diff
@@ -194,5 +195,12 @@ apiRoutes.route('/articles/:slug/comments/:id')
     CommentController.deleteComment
   );
 
+apiRoutes.route('/comments/:id/like')
+  .put(
```
Comment (Dubby20) [inline routes/v1.js]: Can you please explain why you're using a put method here instead of post?

**Review by codinger41 — APPROVED** [review]

File: routes/v1.js (hunk 2 above)
Comment (olajide1234) [reply routes/v1.js]: Updated to Post method as discusses.

File: helpers/isCommentLikedByUser.js (hunk 1 above)
Comment (olajide1234) [reply helpers/isCommentLikedByUser.js]: Updated, thanks

**Review by Dubby20 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Authors Haven - A Social platform for the creative at heart.
 
-[![Build Status](https://travis-ci.com/andela/vidar-ah-backend.svg?branch=develop)](https://travis-ci.com/andela/vidar-ah-backend) [![Coverage Status](https://coveralls.io/repos/github/andela/vidar-ah-backend/badge.svg?branch=develop)](https://coveralls.io/github/andela/vidar-ah-backend?branch=develop) [![](https://img.shields.io/badge/Protected_by-Hound-a873d1.svg)](https://houndci.com)
+[![Build Status](https://travis-ci.com/andela/vidar-ah-backend.svg?branch=develop)](https://travis-ci.com/andela/vidar-ah-backend) [![Coverage Status](https://coveralls.io/repos/github/andela/vidar-ah-backend/badge.svg?branch=develop)](https://coveralls.io/github/andela/vidar-ah-backend?branch=ft-create-user-article-%23164139686) [![](https://img.shields.io/badge/Protected_by-Hound-a873d1.svg)](https://houndci.com)
```
Comment (9jaswag) [inline README.md]: @olajide1234 why are you changing the branch here?

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: controllers/comment.js [hunk 4]
```diff
@@ -152,4 +163,46 @@ export default class CommentController {
...
+* @memberof CommentController
+*
+* @returns {string} Like/unlike comment result
+*/
+  static async likeComment(req, res) {
+    const { id } = req.params;
+    const { id: userId } = req.user;
+    try {
+      // Delete comment like, if already liked by user
+      if (await isCommentLikedByUser(id, userId)) {
```
Comment (9jaswag) [inline controllers/comment.js]: @olajide1234 there might be a Sequelize method to achieve this.

File: README.md (hunk 3 above)
Comment (olajide1234) [reply README.md]: The code coverage results were configured for the `ft-create-user-article-23164139686` branch alone before. @kleva-j advised updating this to make sure the coverage result is for the develop branch.

File: controllers/comment.js (hunk 4 above)
Comment (olajide1234) [reply controllers/comment.js]: I have updated the code to use a Sequelize method.

File: README.md (hunk 3 above)
Comment (9jaswag) [reply README.md]: @olajide1234 I don't quite understand what you mean here. code coverage should work for all branches when configured. The branch referenced in the README dictates which branch's coverage is shown.

Comment (olajide1234) [reply README.md]: Updated and resolved now. Thanks.

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**